### DISCUSSION TITLE
Data Security ?

### DISCUSSION THREAD

**Discussion by kais2021** [rizwansoaib/whatsapp-monitor#103] [General]
Hi!

Great project, great idea. But I miss a whole part of the open source project: The software part for the Server application.

You did not mention on any place, that ALL installations will send ALL DATA to YOUR server, as seen here in your Chrome extension:

-----------------------------
function save(user,t1,t2,t){
  var d   = new Date();
  var curd=d.toLocaleDateString("en-GB").split(' ')[0]
  user=user.replace(/[^a-zA-Z0-9]/g, "")
  curd=curd.replace(/[^a-zA-Z0-9]/g, "")

  const surl='https://whatsappanalysis.in/save/'+user+'/'+curd+'/'+t1+'/'+t2+'/'+t
  var xhr = new XMLHttpRequest();
   xhr.open("GET",surl);
  xhr.send()
  
}
-------------------------

Your website "https://whatsappanalysis.in" is very slow and I didn´t have a good feeling to send my personal and contacts  data to your website. Why you collect such a lot of data???

So is it possible to share the Server code, so everyone can set up his own collecting server and run it on own hardware?

Regards,
Kai

**Comment by rizwansoaib** [comment]
You can use open source code remove const surl to function save () and use
In readme it's mention already

Access History 📡 on web


it's demand of user that we should able to see statistics and previous
history on web so many user is not a developer.

Developer can use open source code and edit them or just paste your
endpoint url.




Website is slow due to its running on AWS free tier soon we will move to
paid server with my personal money.


if you don't want to send your data to server just use open source code.



Thanks & regards
RIZWAN AHMAD

On Thu 12 Aug, 2021, 1:45 AM kais2021, ***@***.***> wrote:

> Hi!
>
> Great project, great idea. But I miss a whole part of the open source
> project: The software part for the Server application.
>
> You did not mention on any place, that ALL installations will send ALL
> DATA to YOUR server, as seen here in your Chrome extension:
> ------------------------------
>
> function save(user,t1,t2,t){
> var d = new Date();
> var curd=d.toLocaleDateString("en-GB").split(' ')[0]
> user=user.replace(/[^a-zA-Z0-9]/g, "")
> curd=curd.replace(/[^a-zA-Z0-9]/g, "")
>
> const surl='
> https://whatsappanalysis.in/save/'+user+'/'+curd+'/'+t1+'/'+t2+'/'+t
> var xhr = new XMLHttpRequest();
> xhr.open("GET",surl);
> xhr.send()
> }
>
> Your website "https://whatsappanalysis.in" is very slow and I didn´t have
> a good feeling to send my personal and contacts data to your website. Why
> you collect such a lot of data???
>
> So is it possible to share the Server code, so everyone can set up his own
> collecting server and run it on own hardware?
>
> Regards,
> Kai
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/rizwansoaib/whatsapp-monitor/discussions/103>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHC2EZHCGN7BJJCTCUSELF3T4LK7HANCNFSM5B7MD5HA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>

-- 
*Disclaimer: *
Messages from 'knit.ac.in <http://knit.ac.in>' mail server 
and any associated files may contain information that is confidential, 
subject to copyright or contains proprietary information. If you are not 
the intended addressee you are hereby notified that any dissemination, 
copying or distribution of these messages, or files associated with these 
messages is prohibited. If you have received the message in error, please 
notify us by replying to the concerned message and deleting it from your 
computer. Please also scan the message to verify that it is virus free. 
Thank you for your time.


This email is governed by the Disclaimer Terms 
of  KNIT.

**Comment by rizwansoaib** [comment]
We take data only minimal which is required


Number, start time, end time, duration



we don't know about which number is using WhatsApp web or what is their
identity but on website you have to know  just number.



May people uses multiple instances (if one instance fail other are running)
and running 24/7 data is integrated at single places and they are fetching
with API and use for own data analysis

We already public our API but as soon as heavy traffic on server I just
remove api endpoint from endpoint and change url.



When we will go for paid server we can provide API for developer for create
your own statistics platform.


if you don't want to send data to our server please build your own api
endpoint someone already created an issue there will be code 10 lines for
php.


And use open source code...

On Thu 12 Aug, 2021, 1:45 AM kais2021, ***@***.***> wrote:

> Hi!
>
> Great project, great idea. But I miss a whole part of the open source
> project: The software part for the Server application.
>
> You did not mention on any place, that ALL installations will send ALL
> DATA to YOUR server, as seen here in your Chrome extension:
> ------------------------------
>
> function save(user,t1,t2,t){
> var d = new Date();
> var curd=d.toLocaleDateString("en-GB").split(' ')[0]
> user=user.replace(/[^a-zA-Z0-9]/g, "")
> curd=curd.replace(/[^a-zA-Z0-9]/g, "")
>
> const surl='
> https://whatsappanalysis.in/save/'+user+'/'+curd+'/'+t1+'/'+t2+'/'+t
> var xhr = new XMLHttpRequest();
> xhr.open("GET",surl);
> xhr.send()
> }
>
> Your website "https://whatsappanalysis.in" is very slow and I didn´t have
> a good feeling to send my personal and contacts data to your website. Why
> you collect such a lot of data???
>
> So is it possible to share the Server code, so everyone can set up his own
> collecting server and run it on own hardware?
>
> Regards,
> Kai
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/rizwansoaib/whatsapp-monitor/discussions/103>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHC2EZHCGN7BJJCTCUSELF3T4LK7HANCNFSM5B7MD5HA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>

-- 
*Disclaimer: *
Messages from 'knit.ac.in <http://knit.ac.in>' mail server 
and any associated files may contain information that is confidential, 
subject to copyright or contains proprietary information. If you are not 
the intended addressee you are hereby notified that any dissemination, 
copying or distribution of these messages, or files associated with these 
messages is prohibited. If you have received the message in error, please 
notify us by replying to the concerned message and deleting it from your 
computer. Please also scan the message to verify that it is virus free. 
Thank you for your time.


This email is governed by the Disclaimer Terms 
of  KNIT.

**Comment by kais2021** [comment]
Hello Rizwan,

Thank you for your detailed and explanatory answer.

From my point of view, this is not an open source project. You only disclose the code of the clients, but without your secret server software the project is not executable.

That's what you're describing.

It came to my attention because the client doesn't work for me. After a little debugging I found out that the data you mentioned is sent to your server. The only thing is that your server is not answering. I have waiting times of more than 30 seconds (!) until maybe an answer comes from there. Very often there are disconnections.

Therefore my suggestion: Convert your server into a paid model and take contributions from users, so that you can afford a good hosting.

Or offer licenses of the server source code, so that people have the chance to build their own server.

The more successful your project becomes, the slower your server will become. Logically, it makes no sense for YOU to store millions of data records of worldwide WhatsApp users on your server permanently.

By the way, it is against the "Basic Data Protection Regulation" which is valid in all countries of the European Union.

Last but not least: You should point out more clearly that your system permanently transfers all data to your server. I think most people who use it are not aware of this.

I wish you much success with your project.

Many greetings,
Kai

**Reply by rizwansoaib** [reply]
I consider your suggestion.
soon we will update readme 

Thank You

**Reply by rizwansoaib** [reply]
I updated access history on web( saved data on server) in readme .



Your more suggestions are welcome @kais2021

**Comment by kais2021** [comment]
Hi!

Please don't get me wrong. It is a great project. I would just like to have an option to keep the data sovereignty myself, especially because your server often has failures.

Greetings,
Kai

**Comment by kais2021** [comment]
Please: You really really need a faster Serer. I wait 30sec to 60secs for one view. 50% of calls are broken
